src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -229,12 +234,37 @@ public void validate(ClientState state)
         validateDefaultTimeToLive(attrs.asNewTableParams());
     }
 
+    private void cloneIndexesIfNeeded(TableMetadata.Builder builder, TableMetadata sourceTableMeta, KeyspaceMetadata targetKeyspaceMeta)
+    {
+        if (createLikeOption == CreateLikeOption.INDEX)
+        {
+            Indexes.Builder idxBuilder = Indexes.builder();
+            for (IndexMetadata indexMetadata : sourceTableMeta.indexes)
```

Comment:
What if `sourceTableMeta.indexes` is empty? What source does not have any index? Then you are creating `idxBuilder` unnecessarily. 

I would expand on the previous suggestion and did:

````
if (createLikeOption != CreateLikeOption.INDEX || sourceTableMeta.indexes.isEmpty())
    return;

...
````